xray/src/utils.rs

```diff
@@ -0,0 +1,8 @@
+use quadtree::NodeId;
+use std::path::{Path, PathBuf};
+
+pub fn get_image_path(directory: &Path, id: NodeId) -> PathBuf {
+    let mut rv = directory.join(&id.to_string());
```

Comment:
nit: you probably don't need the ampersand.
nit2: I think we can simplify it to `directory.join(id.to_string()).with_extension("png")`

also, maybe it would make sense if `get_image_path` accepted a third argument: `extension`?

Reply:
I simplified the method, but kept the hard coded extension.